### ISSUE TITLE
cy.getCookie("name")

### ISSUE THREAD

**Issue by mstruensee** [NicholasBoll/cypress-pipe#24]
I am trying to do ...
`cy.getCookie('xxxxx').should('have.property', 'value', 'xxxxxx')`

but it get set later in the application (page load)
so it fails when cypress looks for it ... so I figured I could try to use pipe here ...

i got it to work by doing ..
`cy.window().pipe(window => window.document.cookie).should('have.property', 'value', 'xxxxxx')`
only problem is here, its a string vs cookie object ... I would like to keep it as the cookie object so I can assert other properties of the cookie (like domain or something) ...

any suggestions?

edit: for now i have this, the pipe forces the cookie to exist for the getCookie command to return it 
```
const getCookies = window => window.document.cookie
	
cy
   .window()
   .pipe(getCookies)
   .should("match", /myCookie=myCookieValue/i)
   .getCookie("myCookie")
   .should("have.property", "domain", "localhost")
```

**Comment by NicholasBoll** [comment]
You could update `getCookies` to look like this:
```js
const getCookies = window => {
  return window.document.cookie.split('; ').reduce((result, item) => {
    const [key, value] = item.split('=')
    result[key] = value
    return result
  }, {})
}

cy
   .window()
   .pipe(getCookies)
   .should("have.property", "domain", "localhost")
```

**Comment by NicholasBoll** [comment]
Cypress may have more access to cookies than `document.cookie` exposes, so pipe might not be able to do exactly what you want. The function I added will turn the `document.cookie` into an object of key/value pairs. The `HTTPOnly` flag will not allow JavaScript to even see a cookie and I noticed the returned object doesn't have all metadata associated with it like `cy.getCookies` does, but perhaps that will work for you?
